From f222de4f0c1c837b6812a176f4f1299b3c44e84b Mon Sep 17 00:00:00 2001
From: Xiao Chen <xiao@apache.org>
Date: Mon, 16 Oct 2017 10:33:16 -0700
Subject: [PATCH 2658/2848] HDFS-12642. Log block and datanode details in
 BlockRecoveryWorker.

(cherry picked from commit 971b3ca1acc15d9a1f8e774291813dafd16ef704)
(cherry picked from commit 6a55d0e67194f1e7a3a23c9bdc263de37d413142)

 Conflicts:
	hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/BlockRecoveryWorker.java

Change-Id: I65b3babb1d90f01095d4b959bb1a71ae9f5da66e
---
 .../hdfs/server/datanode/BlockRecoveryWorker.java  |   13 ++++++++-----
 1 file changed, 8 insertions(+), 5 deletions(-)

diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/BlockRecoveryWorker.java b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/BlockRecoveryWorker.java
index 0fbe2a3..d0ff261 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/BlockRecoveryWorker.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/BlockRecoveryWorker.java
@@ -181,10 +181,8 @@ void syncBlock(List<BlockRecord> syncList) throws IOException {
           getActiveNamenodeForBP(block.getBlockPoolId());
 
       long recoveryId = rBlock.getNewGenerationStamp();
-      if (LOG.isDebugEnabled()) {
-        LOG.debug("block=" + block + ", (length=" + block.getNumBytes()
-            + "), syncList=" + syncList);
-      }
+      LOG.info("BlockRecoveryWorker: block=" + block + " (length="
+          + block.getNumBytes() + "), syncList=" + syncList);
 
       // syncList.isEmpty() means that all data-nodes do not have the block
       // or their replicas have 0 length.
@@ -270,6 +268,11 @@ void syncBlock(List<BlockRecord> syncList) throws IOException {
         break; // we have 'case' all enum values
       }
 
+      LOG.info("BlockRecoveryWorker: block=" + block + " (length="
+          + block.getNumBytes() + "), bestState=" + bestState.name()
+              + ", newBlock=" + newBlock + " (length=" + newBlock.getNumBytes()
+              + "), participatingList=" + participatingList);
+
       List<DatanodeID> failedList = new ArrayList<>();
       final List<BlockRecord> successList = new ArrayList<>();
       for (BlockRecord r : participatingList) {
@@ -320,7 +323,7 @@ private static void logRecoverBlock(String who, RecoveringBlock rb) {
     ExtendedBlock block = rb.getBlock();
     DatanodeInfo[] targets = rb.getLocations();
 
-    LOG.info(who + " calls recoverBlock(" + block
+    LOG.info("BlockRecoveryWorker: " + who + " calls recoverBlock(" + block
         + ", targets=[" + Joiner.on(", ").join(targets) + "]"
         + ", newGenerationStamp=" + rb.getNewGenerationStamp()
         + ")");
-- 
1.7.9.5

